### DISCUSSION TITLE
Are there any plans to support Apache Hudi?

### DISCUSSION THREAD

**Discussion by brucewkz** [duckdb/duckdb#10743] [Ideas]
Currently, many enterprises use Apache Hudi as the data lake storage，so, is there any plan to support it? Thanks.

**Comment by alberttwong** [comment]
Support Hive MetaStore and reading Apache Hudi

Specifically support the following type of queries:
COW Snapshot Queries
COW Incremental Queries
COW Incremental Queries (CDC)
COW Bootstrap Queries
MOR Snapshot Queries
MOR Read-Optimized Queries
MOR Incremental Queries
MOR Incremental Queries (CDC)
MOR Bootstrap queries(RO)
MOR Bootstrap queries(snapshot)
Time travel

**Reply by brucewkz** [reply]
Wonderful! When will it be released?

**Reply by geoHeil** [reply]
https://github.com/apache/hudi-rs could be a first step similar to delta with delta kernel allowing someone to build an integration of duckdb and Hudi

**Reply by geoHeil** [reply]
here a bit ducktaped using python https://dipankar-tnt.medium.com/hudi-rs-with-duckdb-polars-daft-datafusion-single-node-lakehouse-347ee1a45371 but still all in arrow memory.

**Reply by geoHeil** [reply]
though support is read-only -- not for writing (yet)

**Comment by geoHeil** [comment]
How could duckdb perhaps also support the incremental processing capabilities of Hudi?

**Comment by alberttwong** [comment]
Another stop gap option is to use Apache xTable to convert Apache Hudi tables to Delta Lake or Apache Iceberg tables (duckdb already understands delta and iceberg).
